odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/AlertRepositoryImpl.java

```diff
@@ -119,6 +125,68 @@
             .collect(Collectors.toList());
     }
 
+    @Override
+    public Page<AlertDto> listDependentObjectsAlerts(final int page, final int size, final long ownerId) {
+        final List<String> ownOddrns = getObjectsOddrnsByOwner(ownerId);
+
+        final CommonTableExpression<Record1<String>> cte = getChildOddrnsLinageByOwnOddrnsCte(ownOddrns);
+
+        final List<Field<?>> selectFields = Stream
+            .of(ALERT.fields(),
+                DATA_ENTITY.fields(),
+                DATA_ENTITY_SUBTYPE.fields())
+            .flatMap(Arrays::stream)
+            .collect(Collectors.toList());
+
+        final List<AlertDto> data = dslContext.with(cte)
+            .select(selectFields)
+            .select(jsonArrayAgg(field(DATA_ENTITY_TYPE.asterisk().toString())).as(AGG_TYPES_FIELD))
+            .from(DATA_ENTITY)
+            .join(cte.getName()).on(field(cte.getName() + ".child_oddrn", String.class).eq(DATA_ENTITY.ODDRN))
+            .join(ALERT).on(DATA_ENTITY.ODDRN.eq(ALERT.DATA_ENTITY_ODDRN))
+            .join(TYPE_ENTITY_RELATION).on(TYPE_ENTITY_RELATION.DATA_ENTITY_ID.eq(DATA_ENTITY.ID))
+            .join(DATA_ENTITY_TYPE).on(DATA_ENTITY_TYPE.ID.eq(TYPE_ENTITY_RELATION.DATA_ENTITY_TYPE_ID))
+            .join(DATA_ENTITY_SUBTYPE).on(DATA_ENTITY_SUBTYPE.ID.eq(DATA_ENTITY.SUBTYPE_ID))
+            .where(DATA_ENTITY.ODDRN.notIn(ownOddrns))
+            .groupBy(selectFields)
+            .fetchStream()
+            .map(this::mapRecord)
+            .collect(Collectors.toList());
+
+        return Page.<AlertDto>builder()
+            .data(data)
+            .hasNext(true)
+            .total(0L)
+            .build();
+    }
+
+    private List<String> getObjectsOddrnsByOwner(final long ownerId) {
+        return dslContext.select(DATA_ENTITY.ODDRN)
+            .from(DATA_ENTITY)
+            .leftJoin(OWNERSHIP).on(DATA_ENTITY.ID.eq(OWNERSHIP.DATA_ENTITY_ID))
+            .where(OWNERSHIP.OWNER_ID.eq(ownerId))
+            .fetchStreamInto(String.class)
+            .collect(Collectors.toList());
+    }
+
+    private CommonTableExpression<Record1<String>> getChildOddrnsLinageByOwnOddrnsCte(final List<String> ownOddrns) {
+        final Name cteName = name("t");
+
+        final CommonTableExpression<Record> cte = cteName.as(dslContext
+            .select(LINEAGE.asterisk())
+            .from(LINEAGE)
+            .where(LINEAGE.CHILD_ODDRN.in(ownOddrns))
+            .unionAll(dslContext
+                .select(LINEAGE.asterisk())
+                .from(LINEAGE)
+                .join(cteName).on(LINEAGE.CHILD_ODDRN.eq(field("t.parent_oddrn", String.class)))));
+
+        return name("t2")
+            .as(dslContext.withRecursive(cte)
+                .selectDistinct(field(cte.getName() + ".child_oddrn", String.class))
```

Comment:
it's better to use field(name("table_name", "column_name")). Thus you can avoid String column names

Reply:
Let's not use strings at all, we have jOOQ constants. This way we can be sure the app won't compile if we change lineage fields